Mathlib/RepresentationTheory/Homological/GroupHomology/LongExactSequence.lean

```diff
@@ -0,0 +1,158 @@
+
+lemma map_chainsFunctor_shortExact :
+    ShortExact (X.map (chainsFunctor k G)) :=
+  letI := hX.mono_f
+  HomologicalComplex.shortExact_of_degreewise_shortExact _ fun i => {
+    exact := by
+      have : LinearMap.range X.f.hom.hom = LinearMap.ker X.g.hom.hom :=
+        (hX.exact.map (forget₂ (Rep k G) (ModuleCat k))).moduleCat_range_eq_ker
+      simp [moduleCat_exact_iff_range_eq_ker, ker_mapRange, range_mapRange_linearMap X.f.hom.hom
+        (LinearMap.ker_eq_bot.2 <| (ModuleCat.mono_iff_injective _).1 _), this]
+    mono_f := chainsMap_id_f_map_mono X.f i
+    epi_g := letI := hX.epi_g; chainsMap_id_f_map_epi X.g i }
+
+open HomologicalComplex.HomologySequence
+
+/-- The short complex  `Hᵢ(G, X₃) ⟶ Hⱼ(G, X₁) ⟶ Hⱼ(G, X₂)` associated to an exact sequence
+of representations `0 ⟶ X₁ ⟶ X₂ ⟶ X₃ ⟶ 0`. -/
+noncomputable abbrev mapShortComplex₁ {i j : ℕ} (hij : j + 1 = i) :=
+  (snakeInput (map_chainsFunctor_shortExact hX) _ _ hij).L₂'
+
+variable (X) in
+/-- The short complex `Hᵢ(G, X₁) ⟶ Hᵢ(G, X₂) ⟶ Hᵢ(G, X₃)` associated to a short complex of
+representations `X₁ ⟶ X₂ ⟶ X₃`. -/
+noncomputable abbrev mapShortComplex₂ (i : ℕ) := X.map (functor k G i)
+
+/-- The short complex `Hᵢ(G, X₂) ⟶ Hᵢ(G, X₃) ⟶ Hⱼ(G, X₁)` associated to an exact sequence of
+representations `0 ⟶ X₁ ⟶ X₂ ⟶ X₃ ⟶ 0`. -/
+noncomputable abbrev mapShortComplex₃ {i j : ℕ} (hij : j + 1 = i) :=
+  (snakeInput (map_chainsFunctor_shortExact hX) _ _ hij).L₁'
+
+/-- Exactness of `Hᵢ(G, X₃) ⟶ Hⱼ(G, X₁) ⟶ Hⱼ(G, X₂)`. -/
+lemma mapShortComplex₁_exact {i j : ℕ} (hij : j + 1 = i) :
+    (mapShortComplex₁ hX hij).Exact :=
+  (map_chainsFunctor_shortExact hX).homology_exact₁ i j hij
+
+/-- Exactness of `Hᵢ(G, X₁) ⟶ Hᵢ(G, X₂) ⟶ Hᵢ(G, X₃)`. -/
+lemma mapShortComplex₂_exact (i : ℕ) :
+    (mapShortComplex₂ X i).Exact :=
+  (map_chainsFunctor_shortExact hX).homology_exact₂ i
+
+/-- Exactness of `Hᵢ(G, X₂) ⟶ Hᵢ(G, X₃) ⟶ Hⱼ(G, X₁)`. -/
+lemma mapShortComplex₃_exact {i j : ℕ} (hij : j + 1 = i) :
+    (mapShortComplex₃ hX hij).Exact :=
+  (map_chainsFunctor_shortExact hX).homology_exact₃ i j hij
+
+/-- The connecting homomorphism `Hᵢ(G, X₃) ⟶ Hⱼ(G, X₁)` associated to an exact sequence
+`0 ⟶ X₁ ⟶ X₂ ⟶ X₃ ⟶ 0` of representations. -/
+noncomputable abbrev δ (i j : ℕ) (hij : j + 1 = i) :
+    groupHomology X.X₃ i ⟶ groupHomology X.X₁ j :=
+  (map_chainsFunctor_shortExact hX).δ i j hij
+
+open Limits
+
+theorem epi_δ_of_isZero (n : ℕ) (h : IsZero (groupHomology X.X₂ n)) :
+    Epi (δ hX (n + 1) n rfl) := SnakeInput.epi_δ _ h
+
+theorem mono_δ_of_isZero (n : ℕ) (h : IsZero (groupHomology X.X₂ (n + 1))) :
+    Mono (δ hX (n + 1) n rfl) := SnakeInput.mono_δ _ h
+
+theorem isIso_δ_of_isZero (n : ℕ) (hs : IsZero (groupHomology X.X₂ (n + 1)))
+    (h : IsZero (groupHomology X.X₂ n)) :
+    IsIso (δ hX (n + 1) n rfl) := SnakeInput.isIso_δ _ hs h
+
+/-- Given an exact sequence of `G`-representations `0 ⟶ X₁ ⟶f X₂ ⟶g X₃ ⟶ 0`, this expresses an
+`n`-chain `x : Gⁿ →₀ X₁` such that `f ∘ x ∈ Bₙ(G, X₂)` as a cycle. -/
```

Comment:
Isn't the stronger statement that (`f ∘ x ∈ Zₙ(G, X₂)` implies it's a cycle) also true? Is there a reason you wanted this weaker statement?

Reply:
I only stated it for readability of the statement of the next lemma, so I wanted it to be as close to what's needed in that statement as possible. I've added a note to the doc-string saying that + likewise for other similar lemmas.